### PR TITLE
GA-2: initial addition of new base type

### PR DESCRIPTION
**Pull request by sanjayparekh-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/GA-2

### Change description ###
Add new GroupAccess base type

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: repository/src/main/resources/db/migration/V2023100_2__GA-2__CaseAccessGroup.sql
Comment (helenbirdhmcts) [inline repository/src/main/resources/db/migration/V2023100_2__GA-2__CaseAccessGroup.sql:1]: can the formatting be fixed on line 13 and 14?

Comment (abul-syed) [inline repository/src/main/resources/db/migration/V2023100_2__GA-2__CaseAccessGroup.sql:1]: rename flyway script to `V20231030_2__GA-2_CreateCaseAccessGroup.sql `

(correct date + follow previous naming conventions)

File: repository/src/main/resources/db/migration/V2023100_2__GA-2__CaseAccessGroup.sql [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        (select id from field_type where reference = 'Collection' and version = 1 and jurisdiction_id is null),
+        (select id from field_type where reference = 'CaseAccessGroup' and version = 1 and jurisdiction_id is null), '1');
+
+insert into complex_field (reference, label, security_classification, field_type_id, complex_field_type_id)
+values ('caseGroupType', 'CaseGroupType', 'PUBLIC',
+(select id from field_type where reference = 'Text' and version = 1 and jurisdiction_id is null),
+(select id from field_type where reference = 'CaseAccessGroup' and version = 1 and jurisdiction_id is null));
+
+insert into complex_field (reference, label, security_classification, field_type_id, complex_field_type_id)
+values ('caseGroupId', 'CaseGroupId', 'PUBLIC',
```
Comment (helenbirdhmcts) [inline repository/src/main/resources/db/migration/V2023100_2__GA-2__CaseAccessGroup.sql:22]: caseGroupId need to change to caseGroupAccessId as per planning discussion on 14/11

Comment (abul-syed) [reply repository/src/main/resources/db/migration/V2023100_2__GA-2__CaseAccessGroup.sql:22]: `caseAccessGroupId` *

File: repository/src/main/resources/db/migration/V20231311_0000__GA-2_AddCaseAccessGroup.sql [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+and base_field_type_id is null))
+);
+
+INSERT INTO field_type (created_at, reference, base_field_type_id, collection_field_type_id, version)
+VALUES (now(), 'CaseAccessGroups',
+(select id from field_type where reference = 'Collection' and version = 1 and jurisdiction_id is null),
+(select id from field_type where reference = 'CaseAccessGroup' and version = 1 and jurisdiction_id is null), '1');
+
+insert into complex_field (reference, label, security_classification, field_type_id, complex_field_type_id)
+values ('caseGroupType', 'CaseGroupType', 'PUBLIC',
```
Comment (abul-syed) [inline repository/src/main/resources/db/migration/V20231311_0000__GA-2_AddCaseAccessGroup.sql:17]:
```suggestion
values ('caseAccessGroupType', 'CaseAccessGroupType', 'PUBLIC',
```

**Comment by sanjayparekh-hmcts** [comment]
Overtaken by https://github.com/hmcts/ccd-data-store-api/pull/2338

**Pull request closed without merging**